packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/dhtpeer.go

```diff
@@ -963,49 +1455,109 @@ func (dhtPeer *DHTPeer) handleAeaRegisterStream(stream network.Stream) {
 	//linfo().Str("op", "register").
 	//	Msg("Got a new aea register stream")
 
-	clientAddr, err := utils.ReadBytes(stream)
+	buf, err := utils.ReadBytes(stream)
 	if err != nil {
 		lerror(err).Str("op", "register").
-			Msg("while reading client Address from stream")
+			Msg("while reading relay client registration request from stream")
 		err = stream.Reset()
 		ignore(err)
 		return
 	}
 
-	err = utils.WriteBytes(stream, []byte("doneAddress"))
+	msg := &dhtnode.AcnMessage{}
+	err = proto.Unmarshal(buf, msg)
+	if err != nil {
+		lerror(err).Msg("couldn't deserialize acn registration message")
+		// TOFIX(LR) setting Msgs to err.Error is potentially a security vulnerability
+		status := &dhtnode.Status{Code: dhtnode.Status_ERROR_SERIALIZATION, Msgs: []string{err.Error()}}
+		response := &dhtnode.AcnMessage{Version: dhtnode.CurrentVersion, Payload: &dhtnode.AcnMessage_Status{Status: status}}
+		buf, err = proto.Marshal(response)
+		ignore(err)
+		err = utils.WriteBytes(stream, buf)
+		ignore(err)
+		err = stream.Close()
+		ignore(err)
+		return
+	}
+
+	linfo().Msgf("Received relay registration request %s", msg)
+
+	// Get Register message
+	var register *dhtnode.Register
+	switch pl := msg.Payload.(type) {
+	case *dhtnode.AcnMessage_Register:
+		register = pl.Register
+	default:
+		err = errors.New("Unexpected payload")
+		status := &dhtnode.Status{Code: dhtnode.Status_ERROR_UNEXPECTED_PAYLOAD, Msgs: []string{err.Error()}}
+		response := &dhtnode.AcnMessage{Version: dhtnode.CurrentVersion, Payload: &dhtnode.AcnMessage_Status{Status: status}}
+		buf, err = proto.Marshal(response)
+		ignore(err)
+		err = utils.WriteBytes(stream, buf)
+		ignore(err)
+		err = stream.Close()
+		ignore(err)
+		return
+	}
+	record := register.Record
+	clientAddr := record.Address
+
+	//linfo().Msgf("connection from %s established for Address %s",
+	//	stream.Conn().RemotePeer().Pretty(), clientAddr)
+
+	// check if the PoR is valid
+	clientPubKey, err := utils.FetchAIPublicKeyFromPubKey(stream.Conn().RemotePublicKey())
 	ignore(err)
+	status, err := dhtnode.IsValidProofOfRepresentation(record, record.Address, clientPubKey)
```

Comment:
[STP2] peer will verify agent PoR before registering it the agent and saves it locally.
Relay client  node cannot register itself to the dht